servicetalk-opentracing-zipkin-publisher/src/main/java/io/servicetalk/opentracing/zipkin/publisher/reporter/UdpReporter.java

```diff
@@ -0,0 +1,229 @@
+            this.codec = requireNonNull(codec);
+            return this;
+        }
+
+        /**
+         * Sets an IoExecutor to use for writing to the datagram channel.
+         *
+         * @param ioExecutor IoExecutor to use to write with.
+         * @return {@code this}
+         */
+        public Builder ioExecutor(IoExecutor ioExecutor) {
+            this.ioExecutor = requireNonNull(ioExecutor);
+            return this;
+        }
+
+        /**
+         * Enables wire-logging for udp packets sent.
+         *
+         * @param loggerName The name of the logger to log wire events.
+         * @return {@code this}
+         */
+        public Builder enableWireLogging(String loggerName) {
+            this.loggerName = loggerName;
+            return this;
+        }
+
+        /**
+         * Builds a new {@link UdpReporter} instance with this builder's options.
+         *
+         * @return a new {@link UdpReporter}
+         */
+        public UdpReporter build() {
+            return new UdpReporter(this);
+        }
+    }
+
+    private Bootstrap buildBootstrap(EventLoopGroup group, Codec codec, SocketAddress collectorAddress,
+                                     @Nullable String loggerName) {
+        if (!(collectorAddress instanceof InetSocketAddress)) {
+            throw new IllegalArgumentException("collectorAddress " + collectorAddress +
+                    " is invalid for UDP");
+        }
+        return new Bootstrap()
+                .group(group)
+                .channel(datagramChannel(group))
+                .option(RCVBUF_ALLOCATOR, DEFAULT_RECV_BUF_ALLOCATOR)
+                .handler(new ChannelInitializer<Channel>() {
+                    @Override
+                    protected void initChannel(final Channel ch) {
+                        if (loggerName != null) {
+                            ch.pipeline().addLast(new LoggingHandler(loggerName));
+                        }
+                        ch.pipeline().addLast(new ChannelOutboundHandlerAdapter() {
+                            @Override
+                            public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) {
+                                if (msg instanceof Span) {
+                                    byte[] bytes = codec.encoder.encode((Span) msg);
+                                    ByteBuf buf = ctx.alloc().buffer(bytes.length).writeBytes(bytes);
+                                    ctx.write(new DatagramPacket(buf, (InetSocketAddress) collectorAddress), promise);
+                                } else {
+                                    ctx.write(msg, promise);
+                                }
+                            }
+                        });
+                    }
+                });
+    }
+
+    /**
+     * Non-blocking report method.
+     *
+     * @param span the span to report
+     */
+    @Override
+    public void report(final Span span) {
+        if (closed) {
```

Comment:
Once you change `closeAsync()` below to close the `channel`, you can just use `channel.isActive()`.